homeassistant/components/lutron/fan.py

```diff
@@ -0,0 +1,89 @@
+"""Support for Lutron fans."""
+
+from homeassistant.components.fan import DOMAIN, SUPPORT_SET_SPEED, FanEntity
+from homeassistant.util.percentage import (
+    ordered_list_item_to_percentage,
+    percentage_to_ordered_list_item,
+)
+
+from . import DOMAIN as LUTRON_DOMAIN, LUTRON_CONTROLLER, LUTRON_DEVICES, LutronDevice
+
+FAN_OFF, FAN_LOW, FAN_MEDIUM, FAN_MEDIUM_HIGH, FAN_HIGH = 0, 25, 50, 75, 100
+
+DEFAULT_ON_PERCENTAGE = 50
+ORDERED_NAMED_FAN_SPEEDS = [FAN_LOW, FAN_MEDIUM, FAN_MEDIUM_HIGH, FAN_HIGH]
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None) -> None:
+    """Set up the Lutron fans."""
+    devs = []
+
+    # Add Lutron Fans
+    for (area_name, device) in hass.data[LUTRON_DEVICES][DOMAIN]:
+        dev = LutronFan(area_name, device, hass.data[LUTRON_CONTROLLER])
+        devs.append(dev)
+
+    add_entities(devs, True)
+
+
+class LutronFan(LutronDevice, FanEntity):
+    """Representation of a Lutron Fan controller. Including Fan Speed."""
+
+    def __init__(self, area_name, lutron_device, controller):
+        """Initialize the fan device."""
+        self._prev_speed = None
+        super().__init__(area_name, lutron_device, controller)
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {"lutron_integration_id": self._lutron_device.id}
```

Comment:
Ideally this is provided by `LutronDevice`

Reply:
Good point, was just trying to isolate my changes, but I'll move it there.